lib/auth/webauthn/login.go

```diff
@@ -301,6 +311,17 @@ func (f *loginFlow) finish(ctx context.Context, user string, resp *wantypes.Cred
 		return nil, trace.AccessDenied("the given webauthn session allows reuse, but reuse is not permitted in this context")
 	}
 
+	// Verify (and possibly correct) the user verification requirement.
+	// A mismatch here could indicate a programming error or even foul play.
+	uv, _ := userVerificationToProtocol(requiredExtensions.UserVerificationRequirement)
+	if (discoverableLogin || uv == protocol.VerificationRequired) && sd.UserVerification != string(protocol.VerificationRequired) {
```

Comment:
Because some authenticators will do UV=1 regardless, so there is a chance the assertion might still work out.

We capture that in this test: https://github.com/gravitational/teleport/pull/37963/files#diff-93d7067b5fb7f88259f85ac3c3b691bd7f450a5a77e0199d0003a22aff31a474R938-R949